deployment/ccip/changeset/cs_chain_contracts.go

```diff
@@ -351,6 +353,283 @@ func UpdateOnRampsDestsChangeset(e deployment.Environment, cfg UpdateOnRampDests
 	}}, nil
 }
 
+type UpdateOnRampDynamicConfig struct {
+	// UpdatesByChain is a mapping of source -> update.
+	UpdatesByChain map[uint64]OnRampDynamicConfigUpdate
+	// Disallow mixing MCMS/non-MCMS per chain for simplicity.
+	// (can still be achieved by calling this function multiple times)
+	MCMS *MCMSConfig
+}
+
+type OnRampDynamicConfigUpdate struct {
+	FeeQuoter          common.Address
+	MessageInterceptor common.Address
+	FeeAggregator      common.Address
+	AllowlistAdmin     common.Address
+}
+
+func (cfg UpdateOnRampDynamicConfig) Validate() error {
+	for chainSel, config := range cfg.UpdatesByChain {
```

Comment:
Few things that we need to also validate:
- Chain selector is valid
- Chain selector exists in the environment
- OnRamp exists in state
- Timelock & Proposer MCM exist in state
- OnRamp is owned by the Timelock / deployerKey (whichever is making the action)

See the validations for token admin registry changesets: https://github.com/smartcontractkit/chainlink/blob/cd4c9b1a255616fd2407c67440618ea11087fb0a/deployment/ccip/changeset/token_pools.go#L198